### PR TITLE
Added tutorial notebook for using Cleanlab and Pinecone for more reli…

### PR DESCRIPTION
**Pull request by mturk24**
Added a notebook tutorial showcasing how Cleanlab and Pinecone can be used together to create a more reliable, curated, and accurate RAG system. Cory (from Pinecone side) and Matt (from Cleanlab side) spent time iterating on this and the plan is to feature this tutorial in a blog post that will be launched shortly after this is merged

### CODE REVIEW

File: learn/generation/better-rag/Cleanlab_Pinecone_Reliable_Improved_RAG.ipynb [hunk 1]
```diff
@@ -0,0 +1,2604 @@
+{
```
Comment (jseldess) [inline learn/generation/better-rag/Cleanlab_Pinecone_Reliable_Improved_RAG.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    !pip install pinecone-client==5.0.1 sentence-transformers==3.0.1 cleanlab-studio==2.2.1
</pre><p>We can change <code>pinecone-client</code> to <code>pinecone</code>. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/394/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (jseldess) [inline learn/generation/better-rag/Cleanlab_Pinecone_Reliable_Improved_RAG.ipynb:1]: <p>"This allows us" should be the start of a new sentence.</p><p> </p><p>Also, please change the link to <a href="https://docs.pinecone.io/guides/indexes/understanding-indexes" target="_blank">https://docs.pinecone.io/guides/indexes/understanding-indexes</a>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/394/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (jseldess) [inline learn/generation/better-rag/Cleanlab_Pinecone_Reliable_Improved_RAG.ipynb:1]: <p>I'd edit the last sentence a bit: "This includes the two helper functions,&nbsp;<code>parse_category()</code>&nbsp;and&nbsp;<code>classify()</code>&nbsp;."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pinecone-io/examples/pull/394/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='19'/>

**Review by jseldess — APPROVED** [review]
This looks good to me overall, @mturk24. Just left a few suggestions.

**Comment by mturk24** [comment]
> This looks good to me overall, @mturk24. Just left a few suggestions.

Great thanks @jseldess! I just included edits in the latest commit that address your comments.

**Pull request merged**